gomint-server/src/main/java/io/gomint/server/entity/monster/EntityIronGolem.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020 Gomint team
+ *
+ * This code is licensed under the BSD license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package io.gomint.server.entity.monster;
+
+import io.gomint.inventory.item.ItemIronIngot;
+import io.gomint.inventory.item.ItemRedFlower;
+import io.gomint.math.Location;
+import io.gomint.server.entity.Attribute;
+import io.gomint.server.entity.EntityLiving;
+import io.gomint.server.entity.EntityTags;
+import io.gomint.server.entity.EntityType;
+import io.gomint.server.registry.RegisterInfo;
+import io.gomint.server.world.WorldAdapter;
+
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * @author joserobjr
+ * @since 2021-01-12
+ */
+@RegisterInfo( sId = "minecraft:iron_golem" )
+public class EntityIronGolem extends EntityLiving implements io.gomint.entity.monster.EntityIronGolem {
+    /**
+     * Constructs a new EntityIronGolem
+     *
+     * @param world The world in which this entity is in
+     */
+    public EntityIronGolem( WorldAdapter world ) {
+        super( EntityType.IRON_GOLEM, world );
+        this.initEntity();
+    }
+
+    /**
+     * Create new entity iron golem for API
+     */
+    public EntityIronGolem() {
+        super( EntityType.IRON_GOLEM, null );
+        this.initEntity();
+    }
+
+    private void initEntity() {
+        this.addAttribute(Attribute.HEALTH);
+        this.setMaxHealth(100);
+        this.setHealth(100);
+        this.setSize(1.4f, 2.9f);
+    }
+
+    @Override
+    protected void kill() {
+        super.kill();
+        
+        if (isDead()) {
+            return;
+        }
+        
+        // Item drops
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        Location location = this.getLocation();
+        this.world.dropItem(location, ItemIronIngot.create(random.nextInt(3, 6)));
+        
+        if (isLastDamageCausedByPlayer()) {
```

Comment:
It seems (at least according to the minecraft wiki) that red flowers always drop

Reply:
Nice catch!